src/mui/input/ReferenceArrayInput.js

```diff
@@ -172,7 +175,7 @@ export class ReferenceArrayInput extends Component {
             );
         }
 
-        if (!(referenceRecords && referenceRecords.length > 0) && !allowEmpty) {
```

Comment:
You miss a case where the current record references non-existent records in the distant resource. E.g. an article with a list of author_ids, some of which don't exist in the Author resource. The previous code was there to avoid the app to break in that case (and we should keep this security).